webapp/loggedin/surveyViews/taxonomyEdit/components/taxonomyEdit.js

```diff
@@ -1,89 +1,156 @@
+                 onChange={value => putTaxonomyProp(taxonomy, 'name', StringUtils.normalizeName(value))}
+                 readOnly={readOnly} />
+        </div>
+      </FormItem>
+
+      <div className="button-bar">
+        {
+          !readOnly &&
+          <UploadButton label={i18n.t('common.csvImport')}
+                        disabled={Taxonomy.isPublished(taxonomy)}
+                        accept=".csv"
+                        onChange={async ([file]) => { await uploadTaxonomyFile(taxonomy, file) } }/>
+
+        }
+        <DownloadButton href={`/api/survey/${surveyId}/taxonomies/${Taxonomy.getUuid(taxonomy)}/export?draft=true`}
+                        disabled={R.isEmpty(taxa)}
+                        label={i18n.t('common.csvExport')} />
+      </div>
+    </div>
+  )
+}
+
+const TaxonomyEdit = props => {
+
+  const {
+    surveyId, taxonomy, taxa,
+    setTaxonomyForEdit,
+    readOnly,
+    putTaxonomyProp,
+    uploadTaxonomyFile,
+    initList,
+    activeJob,
+  } = props
+
   const i18n = useI18n()
 
-  return (
-    <div className="taxonomy-edit">
+  const taxonomyUuid = Taxonomy.getUuid(taxonomy)
 
-      <div className="taxonomy-edit__header">
+  const vernacularLanguageCodes = R.reduce(
+    (acc, taxon) => R.concat(acc, R.difference(R.keys(Taxon.getVernacularNames(taxon)), acc)),
+    [],
+    taxa
+  )
 
-        <ErrorBadge validation={validation}/>
+  useOnUpdate(() => {
+    if (!activeJob) {
+      initList(TaxonomyEditState.keys.taxa, moduleApiUri, moduleApiCountUri)
+    }
+  }, [activeJob])
 
-        <FormItem label={i18n.t('taxonomy.edit.taxonomyName')}>
-          <div>
-            <Input value={Taxonomy.getName(taxonomy)}
-                   validation={Validation.getFieldValidation('name')(validation)}
-                   onChange={value => putTaxonomyProp(taxonomy, 'name', StringUtils.normalizeName(value))}
-                   readOnly={readOnly}/>
-          </div>
-        </FormItem>
-
-        <div className="button-bar">
-          {
-            !readOnly &&
-            <UploadButton label={i18n.t('common.csvImport')}
-                          disabled={Taxonomy.isPublished(taxonomy)}
-                          accept=".csv"
-                          onChange={(files) => uploadTaxonomyFile(taxonomy, files[0])}/>
-
-          }
-          <DownloadButton href={`/api/survey/${surveyId}/taxonomies/${Taxonomy.getUuid(taxonomy)}/export?draft=true`}
-                          disabled={R.isEmpty(taxa)}
-                          label={i18n.t('common.csvExport')}/>
-        </div>
-      </div>
+  const gridTemplateColumns = `.1fr .1fr .2fr .2fr .4fr ${R.isEmpty(vernacularLanguageCodes) ? '' : `repeat(${vernacularLanguageCodes.length}, 60px)`}`
+  const moduleApiUri = `/api/survey/${surveyId}/taxonomies/${taxonomyUuid}/taxa?draft=true&validate=true`
+  const moduleApiCountUri = `/api/survey/${surveyId}/taxonomies/${taxonomyUuid}/taxa/count`
```

Comment:
the apiCount url is automatically generated by tableView by concatenating /count. 
I'd rather pass the params (That can eventually be used also by the rest call count)